### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by ZoeMitchelll**
Just adding authentication before opening up the stats page

### CODE REVIEW

File: capstone/src/main/java/com/google/sps/data/SentimentData.java [hunk 1]
```diff
@@ -3,15 +3,15 @@
 import com.google.cloud.language.v1.Sentiment;
 import java.io.IOException;
 
-public class SentimentData{
+public class SentimentData {
```
Comment (nitinramadoss) [inline capstone/src/main/java/com/google/sps/data/SentimentData.java:6]: I have made edits to this class so that it works on the live server

File: capstone/src/main/webapp/script.js [hunk 2]
```diff
@@ -200,3 +200,14 @@ function loadReponseChart() {
       document.getElementById('response-bar-chart'));
   chart.draw(stats, null);
 }
+
+async function showStats() {
+  const logStatus = await fetch('/status');
+  const status = await logStatus.json();
+  console.log(status + ': status');
+  if (status==false) {
```
Comment (nitinramadoss) [inline capstone/src/main/webapp/script.js:208]: You could rephrase this as if (!status)

File: capstone/src/main/java/com/google/sps/data/SentimentData.java (hunk 1 above)
Comment (ZoeMitchelll) [reply capstone/src/main/java/com/google/sps/data/SentimentData.java:6]: Okay, thanks!

File: capstone/src/main/java/com/google/sps/data/MapData.java [hunk 3]
```diff
@@ -4,42 +4,30 @@
...
-        zipPrecinctMap.put("94129", new ArrayList<>(
-            Arrays.asList("Richmond")));
-        zipPrecinctMap.put("94131", new ArrayList<>(
-            Arrays.asList("Ingleside","Park")));
-    }
-    public ArrayList getPrecincts(String zipcode){
-        return zipPrecinctMap.get(zipcode);
-    }
+    zipPrecinctMap = new HashMap<>();
+    zipPrecinctMap.put("94103", new ArrayList<>(Arrays.asList("Southern", "Mission")));
```
Comment (jasperlee-google) [inline capstone/src/main/java/com/google/sps/data/MapData.java:12]: Nice! Much easier to read :-)

File: capstone/src/main/java/com/google/sps/servlets/CheckStatus.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/status")
+public class CheckStatus extends HttpServlet {
```
Comment (jasperlee-google) [inline capstone/src/main/java/com/google/sps/servlets/CheckStatus.java:14]: Should this be called `CheckUserStatus` to be more explicit about what we are checking the status for?

File: capstone/src/main/webapp/script.js (hunk 2 above)
Comment (jasperlee-google) [reply capstone/src/main/webapp/script.js:208]: +1 this is probably more common to see

File: capstone/src/main/webapp/script.js [hunk 5]
```diff
@@ -207,3 +200,14 @@ function loadReponseChart() {
       document.getElementById('response-bar-chart'));
   chart.draw(stats, null);
 }
+
+async function showStats() {
+  const logStatus = await fetch('/status');
+  const status = await logStatus.json();
+  console.log(status + ': status');
```
Comment (jasperlee-google) [inline capstone/src/main/webapp/script.js:207]: Should this be reversed? (ie: should this log `status: false`).
